File: searchInsertPosition.java
public class Solution {
    public int searchInsert(int[] A, int target) {
        int low = 0;
        int high = A.length-1;
        while (low <= high) {
            int middle = low + (high-low)/2;
            if (A[middle] == target) {
                return middle;
            }
            else if (A[middle] < target) {
                low = middle+1;
            }
            else {
                high = middle -1;
            }
        }
        return low;
    }
}

File: Algorithms/searchInsertPosition.py
class Solution:
    # @param A, a list of integers
    # @param target, an integer to be inserted
    # @return integer
    def searchInsert(self, A, target):
        low, high = 0, len(A)-1
        while low < high:
            mid = low + (high - low)/2
            if A[mid] == target:
                return mid
            elif A[mid] < target:
                low = mid + 1
            else:
                high = mid
        return (low + 1) if A[low] < target else low
